src/blas/KokkosBlas1_iamax.hpp

```diff
@@ -136,7 +136,10 @@ iamax (const RV& R, const XMV& X,
       typename RV::non_const_value_type,
       typename RV::non_const_value_type* >::type,
     typename KokkosKernels::Impl::GetUnifiedLayout<RV>::array_layout,
-    typename RV::device_type,
+    typename Kokkos::Impl::if_c<
```

Comment:
Use `std::conditional` (lives in `<type_traits>`).  It works just like `if_c` here.

Reply:
@mhoemmen I am curious. why don't we just use if_c?

Reply:
@vqd8a I'm pretty sure `if_c` is Kokkos' pre-c++11 implementation of `std::conditional`, and some future cleanup in Kokkos will likely remove `if_c` so if nothing else it saves work later making the change; in general it is better to use standard library implementations when feasible.

Reply:
1. Prefer Standard Library features to Kokkos features that do the same thing.
2. The feature `if_c` has that `std::conditional` lacks is the `select` method.

Reply:
Okay. Thanks @ndellingwood and @mhoemmen for clarifying.

Reply:
@mhoemmen ```std::conditional``` is used now.